cg/cli/set.py

```diff
@@ -39,164 +43,237 @@ def family(context, action, priority, panels, family_id):
-@click.option('-c', '--customer', help='updates customer, input format custXXX.')
-@click.option('-C', '--add-comment', 'comment', type=str, help='adds a note/comment to a sample, '
-                                                               'put text between quotation marks! '
-                                                               'This will not overwrite the '
-                                                               'current comment.')
-@click.option('-d', '--downsampled-to', type=int, help='sets number of downsampled \
-              total reads. Enter 0 to reset.')
-@click.option('-a', '--application-tag', 'apptag', help='sets application tag.')
-@click.option('-k', '--capture-kit', help='sets capture kit.')
-@click.option('--data-analysis', help='sets data-analysis.')
-@click.argument('sample_id')
+@click.option("-s", "--sex", type=click.Choice(["male", "female", "unknown"]))
+@click.option("-c", "--customer", help="updates customer, input format custXXX.")
+@click.option(
+    "-C",
+    "--add-comment",
+    "comment",
+    type=str,
+    help="adds a note/comment to a sample, "
+    "put text between quotation marks! "
+    "This will not overwrite the "
+    "current comment.",
+)
+@click.option(
+    "-d",
+    "--downsampled-to",
+    type=int,
+    help="sets number of downsampled \
+              total reads. Enter 0 to reset.",
+)
+@click.option("-a", "--application-tag", "apptag", help="sets application tag.")
+@click.option("-k", "--capture-kit", help="sets capture kit.")
+@click.option("--data-analysis", help="sets data-analysis.")
+@click.option("-n", "--name", help="sets name.")
+@click.argument("sample_id")
 @click.pass_context
-def sample(context, sex, customer, comment, downsampled_to, apptag, capture_kit,
-           data_analysis, sample_id):
+def sample(
+    context,
+    sex,
+    customer,
+    comment,
+    downsampled_to,
+    apptag,
+    capture_kit,
+    data_analysis,
+    name,
+    sample_id,
+):
     """Update information about a sample."""
-    sample_obj = context.obj['status'].sample(sample_id)
+    sample_obj = context.obj["status"].sample(sample_id)
 
     if sample_obj is None:
-        click.echo(click.style(f"Can't find sample {sample_id}", fg='red'))
+        click.echo(click.style(f"Can't find sample {sample_id}", fg="red"))
         context.abort()
     else:
-        click.echo(click.style(f"Found sample {sample_obj.internal_id}", fg='green'))
+        click.echo(click.style(f"Found sample {sample_obj.internal_id}", fg="green"))
 
     echo_msg = ""
 
     if comment:
-        echo_msg += click.style(f"\nComment added to sample", fg='green')
+        echo_msg += click.style(f"\nComment added to sample", fg="green")
     else:
         comment = ""
 
+    if name:
+        if sample_obj.name != name:
+            echo_msg += click.style(
+                f"\nUpdate sample name: {sample_obj.name} -> {name}", fg="green"
```

Comment:
I think you used to tell me not ti use f-strings for logging because its more time consuming than using %s. And this is some sort of logging? But you probably know best.

Reply:
I think that might be right, I'll leave it like this for now but discuss it with Kenny or Måns